### ISSUE TITLE
`no-unused-vars` warning while type is used in decorators

### ISSUE THREAD

**Issue by Tricked-dev** [denoland/deno_lint#1011]
### Lint Name

no-unused-vars
### Code Snippet

```ts
import type { Options } from './types.ts';

function deco<T>(constructor: Function) {}

@deco<Options>()
export class Example {}
```

```ts
//types.ts
interface Options {
	foo: string;
}
```

### Expected Result

No lint error

### Actual Result

Lint complains about the type not being used

### Additional Info
```
(no-unused-vars) `Options` is never used
import type { Options } from './types.ts';
              ^^^^^^^
    at ./mod.ts:1:14

    hint: If this is intentional, alias it with an underscore like `Options as _Options`
    help: for further information visit https://lint.deno.land/#no-unused-vars
```
### Version

deno 1.18.2 (release, x86_64-unknown-linux-gnu)
v8 9.8.177.6
typescript 4.5.2

**Comment by magurotuna** [comment]
Thanks for filing the issue. Looking into it a little bit, I found that this issue comes from the parser, and that it's already fixed in the latest version of swc. We first need to upgrade swc's version in the crates that deno_lint depends on, so I started from https://github.com/dprint/dprint-swc-ecma-ast-view/pull/34.
